app/models/post.rb

```diff
@@ -281,6 +281,18 @@ def valid_board_section
     errors.add(:section, "must be in the post's board")
   end
 
+  def valid_coauthors
+    return if self.unjoined_authors.empty?
+    return if self.user.blocked_interaction_users(reciever_direction: 'either').empty? && self.joined_authors.empty?
+    self.unjoined_authors.each do |unjoined|
+      next if unjoined.blocked_interaction_users(reciever_direction: 'either').empty?
+      errors.add(:post_author, "cannot be added") unless self.user.can_interact_with?(unjoined)
```

Comment:
I have not mentally parsed this entire loop but I am confident we need a break somewhere